arm9/overlays/24/src/ov24_02254CA0.c

```diff
@@ -0,0 +1,246 @@
+#include "global.h"
+#include "bg_window.h"
+#include "unk_0200CA44.h"
+#include "gf_gfx_loader.h"
+#include "overlay_24.h"
+
+extern BOOL sub_0208946C(u32, void*, u32);
+extern void sub_02089444(u32, void*, u32);
+extern void ov20_02252D7C(u32, u32);
+extern void ov20_02253F28(u32*, u32);
+extern void Poketch_PlaySoundEffect(u32);
+extern void* ov20_022538A0(void*);
+extern void ov20_02253888(u32*, void*);
+extern BOOL ov20_022537B8(void*);
+extern BOOL ov20_02253794(void*, u8);
+extern void ov20_022537E0(const u32*, u32, void*, void*, void*, u32, u32);
+extern BOOL ov20_02253FBC(void*, NarcId, u32, u32, HeapID);
+extern u32* ov20_02253E74(u32, void*, void*);
+extern void ov20_02254014(void*);
+extern void ov20_02253F14(u32, void*);
+extern void ov20_022536F4(void*, u32);
+extern u32 ov20_02252D34();
+extern u32 ov20_02252D24();
+
+BOOL ov24_02254CA0(MemoPadDisplayHandler** displayHandlerOut, MemoPadDrawState* drawState) {
+    MemoPadDisplayHandler* displayHandler = AllocFromHeap(HEAP_ID_POKETCH_APP, sizeof(MemoPadDisplayHandler));
+    if (displayHandler) {
+        GF_ASSERT(GF_heap_c_dummy_return_true(HEAP_ID_POKETCH_MAIN));
+        ov20_022536F4(&(displayHandler->unk8), 0x10);
+        GF_ASSERT(GF_heap_c_dummy_return_true(HEAP_ID_POKETCH_MAIN));
+        displayHandler->drawState = drawState;
+        displayHandler->config = (BgConfig*)ov20_02252D34();
+        displayHandler->unk50 = ov20_02252D24();
+        GF_ASSERT(GF_heap_c_dummy_return_true(HEAP_ID_POKETCH_MAIN));
+        *displayHandlerOut = displayHandler;
+        return TRUE;
+    }
+    return FALSE;
+}
+
+BOOL ov24_02254D00(MemoPadDisplayHandler* displayHandler) {
+    static const WindowTemplate template = {
+        .bgId = GF_BG_LYR_SUB_3,
+        .left = 2,
+        .top = 2,
+        .width = 20,
+        .height = 19,
+        .palette = 0,
+        .baseTile = 12,
+    };
+    displayHandler->window = AllocWindows(HEAP_ID_POKETCH_APP, 1);
+    if (displayHandler->window) {
+        AddWindow(displayHandler->config, displayHandler->window, &template);
+        if (sub_0208946C(displayHandler->drawState->unk16E8, displayHandler->window->pixelBuffer, 0x2f80) == 0) {
+            FillWindowPixelBuffer(displayHandler->window, MEMO_PAD_PIXEL_TYPE_EMPTY);
+        }
+        return TRUE;
+    }
+    return FALSE;
+}
+
+void ov24_02254D48(MemoPadDisplayHandler* displayHandler) {
+    if (displayHandler) {
+        GF_ASSERT(GF_heap_c_dummy_return_true(HEAP_ID_POKETCH_MAIN));
+        if (displayHandler->window) {
+            sub_02089444(displayHandler->drawState->unk16E8, displayHandler->window->pixelBuffer, 0x2f80);
+            RemoveWindow(displayHandler->window);
+            FreeToHeap(displayHandler->window);
+        }
+        FreeToHeap(displayHandler);
+    }
+}
+
+void ov24_02254D8C(MemoPadDisplayHandler* displayHandler, u32 arg1) {
+    // TODO: types
```

Comment:
types?

Reply:
I definitely think its an array of structs and the way it is now is just to get the rodata in there. ``ov20_02253E74`` is undocumented and since I was only looking at ov24 I have no idea what the actual data represents

Reply:
understandable